ledger/catchpointtracker.go

```diff
@@ -1446,21 +1442,55 @@ func accountHashBuilderV6(addr basics.Address, accountData *baseAccountData, enc
 	return finishV6(hash, prehash)
 }
 
-// accountHashBuilderV6 calculates the hash key used for the trie by combining the account address and the account data
-func resourcesHashBuilderV6(addr basics.Address, cidx basics.CreatableIndex, ctype basics.CreatableType, updateRound uint64, encodedResourceData []byte) []byte {
-	hash := hashBufV6(updateRound, byte(ctype+1)) // one or two ( asset / application ) so we could differentiate the hashes.
+// hashKind enumerates the possible data types hashed into a catchpoint merkle
+// trie.  Each merkle trie hash includes the hashKind byte at a known-offset.
+// By encoding hashKind at a known-offset, it's possible for hash readers to
+// disambiguate the hashed resource.
+//go:generate stringer -type=hashKind
+type hashKind byte
+
+// Defines known kinds of hashes.  Changing an enum ordinal value is a
+// breaking change.
+const (
+	accountHK hashKind = iota
+	assetHK
+	appHK
+	kvHK
+)
+
+// HashKindEncodingIndex defines the []byte offset where the hash kind is
+// encoded.
+const HashKindEncodingIndex = 4
+
+func creatableHashKindFromResourcesData(rd resourcesData, a basics.Address, ci basics.CreatableIndex) (hashKind, error) {
+	if rd.IsAsset() {
+		return assetHK, nil
+	} else if rd.IsApp() {
+		return appHK, nil
+	}
+	return accountHK, fmt.Errorf("unknown creatable for addr %s, aidx %d, data %v", a.String(), ci, rd)
+}
+
+// resourcesHashBuilderV6 calculates the hash key used for the trie by combining the creatable's resource data and its index
+func resourcesHashBuilderV6(rd resourcesData, addr basics.Address, cidx basics.CreatableIndex, updateRound uint64, encodedResourceData []byte) ([]byte, error) {
+	hk, err := creatableHashKindFromResourcesData(rd, addr, cidx)
+	if err != nil {
+		return nil, err
+	}
+
+	hash := hashBufV6(updateRound, hk)
 
 	prehash := make([]byte, 8+crypto.DigestSize+len(encodedResourceData))
 	copy(prehash[:], addr[:])
 	binary.LittleEndian.PutUint64(prehash[crypto.DigestSize:], uint64(cidx))
 	copy(prehash[crypto.DigestSize+8:], encodedResourceData[:])
 
-	return finishV6(hash, prehash)
+	return finishV6(hash, prehash), nil
 }
 
 // kvHashBuilderV6 calculates the hash key used for the trie by combining the key and value
 func kvHashBuilderV6(key string, value []byte) []byte {
-	hash := hashBufV6(0, 3) // 3 indicates a kv pair
```

Comment:
`hashBufV6` should use the new constant HashKindEncodingIndex I guess

Reply:
@algorandskiy I'm not following the request - let me know if it's still a pending concern after reading the note below?

The thread highlights _before_ not _after_ the PR.  The corresponding line _is_ changed to use the constant.  So, I don't _think_ there's further action here.

Reply:
>	hash[hashKindEncodingIndex] = byte(kind)
well, I overlooked it somehow. apologies.